### DISCUSSION TITLE
Timeline for New Releases

### DISCUSSION THREAD

**Discussion by bfailing** [artisan-roaster-scope/artisan#976] [General]
Is there a timeline for new releases anywhere?  If not, how is the decision made to issue a new release?

**Comment by poundy** [comment]
Is there something specific you're waiting for? 

I am sure there are many inputs to releases (I'm not part of the team who would decide this) but I don't think there's actually much reason for this to matter. There's the stable releases that come along every so often, and there's the continuous build versions that are released that address minor issues along the way. There's no difference that I'm aware of that make a continuous build version "unattractive" or not backward compatible with the current release, meaning your experience on that should be the same as someone else on the stable release. 

Some of the inputs I imagine are things like securing a release sponsor (as we've seen in the last few releases); timing around actual events to help maximise visibility; significance and completeness of the changes in the application (the larger the change, the longer you'd expect to see that take to develop)

One important point in all this discussion is that the product isn't funded except by those who provide financial support through sponsorship or donation. This is certainly nobody's "day job" to maintain, so it's not like the programming team have clear 8 hour days to continue the development work, so if there's a big feature in development it'll take a longer time to get to release.

**Reply by MAKOMO** [reply]
You are spot on. There are many inputs to consider. Securing a release sponsor is one of this as publishing a release takes quite some extra resources. Luckily we found one by now for the next release. Artisan release are also aligned with external developments (new dependent library releases, release of new roasting machines, development progress of external software like artisan.plus). Another factor is the completion of new features or changes we intended to apply to that release (we maintain a long list and pick those we have the most interest in and see a chance for realization). Once they are all in, we start a phase of testing for stability. Then there is a continuous stream of external bug reports and feature requests which need to be decided upon. We are not releasing anything while we still have some of those pending. Finally, we need to find resource on preparing the release documentation. Overall the decision on a release is a dynamic and complex process and very much depends on the interest and available resources of the development team and some external factors, currently just Dave and myself.

A note on those continuous builds: those are development builds and as such not guaranteed to install and run at all. They also might contain known issues and experimental code as they are work in progress. Further, those builds are not available for all platforms and are not developer signed nor notarised via Apple. However, we try to keep those in a good state and often they come with fixes for issues discovered in the previous release and thus can perform even better than that previous version.
